<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Clever Maze</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedKingdom"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout" rowstretch="1,5" columnstretch="2,1">
    <item row="0" column="1">
     <widget class="QLabel" name="label_Title">
      <property name="font">
       <font>
        <family>Consolas</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: qradialgradient(spread:pad, cx:0.5, cy:0.5, radius:0.5, fx:0.5, fy:0.5, stop:0 rgba(0, 0, 0, 0), stop:0.52 rgba(0, 0, 0, 0), stop:0.565 rgba(82, 121, 76, 33), stop:0.65 rgba(159, 235, 148, 64), stop:0.721925 rgba(255, 238, 150, 129), stop:0.77 rgba(255, 128, 128, 204), stop:0.89 rgba(191, 128, 255, 64), stop:1 rgba(0, 0, 0, 0));</string>
      </property>
      <property name="text">
       <string>Clever Maze</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QToolBox" name="toolBox">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="page_Design">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>277</width>
         <height>373</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Design</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="5" column="0">
         <widget class="QLabel" name="label_Set_As">
          <property name="text">
           <string>Set As</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="comboBox">
          <item>
           <property name="text">
            <string>Accessible</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Inaccessible</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Entrance</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Exit</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Column Count</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinBox_Row">
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="8" column="0" colspan="2">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="2">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="9" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Auto Maze Options</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="1" column="1">
            <widget class="QSpinBox" name="spinBox_3"/>
           </item>
           <item row="4" column="0" colspan="3">
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>Auto Maze Go!</string>
             </property>
             <property name="icon">
              <iconset resource="Icons.qrc">
               <normaloff>:/Menu Icon/Create Video.svg</normaloff>:/Menu Icon/Create Video.svg</iconset>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="spinBox_5"/>
           </item>
           <item row="1" column="2">
            <widget class="QSpinBox" name="spinBox_4"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Entrence at</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QSpinBox" name="spinBox_6"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Exit at</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="spinBox_Column">
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Row Count</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="7" column="0" colspan="2">
         <widget class="QPushButton" name="pushButton_Check">
          <property name="text">
           <string>Check the Maze</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_Game">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>277</width>
         <height>373</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Game</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1">
        <item row="8" column="0">
         <widget class="QPushButton" name="pushButton_Solution">
          <property name="text">
           <string>View Solution</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QSpinBox" name="spinBox_SolutionNo">
          <property name="prefix">
           <string>Solution </string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QCommandLinkButton" name="commandLinkButton_Start">
          <property name="font">
           <font>
            <family>Segoe UI</family>
            <pointsize>12</pointsize>
            <weight>50</weight>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>Start the Game</string>
          </property>
          <property name="description">
           <string/>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QPushButton" name="pushButton_Step">
          <property name="text">
           <string>Step by Step</string>
          </property>
         </widget>
        </item>
        <item row="9" column="1">
         <widget class="QProgressBar" name="progressBar_Step">
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QLabel" name="label_Time_2">
          <property name="text">
           <string>Solution View</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignBottom|Qt::AlignHCenter</set>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QLCDNumber" name="lcdNumber_Step"/>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Step Used</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLCDNumber" name="lcdNumber_Time">
          <property name="smallDecimalPoint">
           <bool>false</bool>
          </property>
          <property name="digitCount">
           <number>5</number>
          </property>
          <property name="value" stdset="0">
           <double>0.000000000000000</double>
          </property>
          <property name="intValue" stdset="0">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_Time">
          <property name="text">
           <string>Time Used</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>546</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile_F">
    <property name="title">
     <string>File(&amp;F)</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionImport"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionConfiguration"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp_H">
    <property name="title">
     <string>Help(&amp;H)</string>
    </property>
    <addaction name="actionEnglish"/>
    <addaction name="actionChinese"/>
   </widget>
   <widget class="QMenu" name="menuMore_O">
    <property name="title">
     <string>More(&amp;O)</string>
    </property>
    <widget class="QMenu" name="menuAbout_Teddy_van_Jerry">
     <property name="title">
      <string>About Teddy van Jerry</string>
     </property>
     <addaction name="actionGitHub"/>
     <addaction name="actionCSDN"/>
     <addaction name="actionStack_Overflow"/>
     <addaction name="actionBilibili"/>
    </widget>
    <widget class="QMenu" name="menuSend_Feedback">
     <property name="title">
      <string>Send Feedback</string>
     </property>
     <addaction name="actionGitHub_2"/>
     <addaction name="actionCSDN_2"/>
    </widget>
    <addaction name="menuAbout_Teddy_van_Jerry"/>
    <addaction name="menuSend_Feedback"/>
    <addaction name="actionGitHub_Repository"/>
    <addaction name="actionVersion"/>
    <addaction name="actionCheck_Update"/>
   </widget>
   <addaction name="menuFile_F"/>
   <addaction name="menuHelp_H"/>
   <addaction name="menuMore_O"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="statusTip">
    <string/>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionImport"/>
   <addaction name="actionSave"/>
   <addaction name="actionClose"/>
   <addaction name="actionExit"/>
   <addaction name="separator"/>
   <addaction name="actionAuto_Maze"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="actionClear"/>
   <addaction name="separator"/>
   <addaction name="actionSolution"/>
   <addaction name="actionSolution_by_Step"/>
   <addaction name="actionStop"/>
   <addaction name="separator"/>
   <addaction name="actionConfiguration"/>
   <addaction name="actionEnglish"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/New.svg</normaloff>:/Menu Icon/New.svg</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Open.svg</normaloff>:/Menu Icon/Open.svg</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Save.svg</normaloff>:/Menu Icon/Save.svg</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/SaveAs.svg</normaloff>:/Menu Icon/SaveAs.svg</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Close.svg</normaloff>:/Menu Icon/Close.svg</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Exit.svg</normaloff>:/Menu Icon/Exit.svg</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Undo.svg</normaloff>:/Menu Icon/Undo.svg</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Redo.svg</normaloff>:/Menu Icon/Redo.svg</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/DeleteImage.svg</normaloff>:/Menu Icon/DeleteImage.svg</iconset>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionSolution">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Create Images in Range.svg</normaloff>:/Menu Icon/Create Images in Range.svg</iconset>
   </property>
   <property name="text">
    <string>Solution</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionSolution_by_Step">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Create Image.svg</normaloff>:/Menu Icon/Create Image.svg</iconset>
   </property>
   <property name="text">
    <string>Solution by Step</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="actionAuto_Maze">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Create Video.svg</normaloff>:/Menu Icon/Create Video.svg</iconset>
   </property>
   <property name="text">
    <string>Auto Maze</string>
   </property>
  </action>
  <action name="actionEnglish">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Help.svg</normaloff>:/Menu Icon/Help.svg</iconset>
   </property>
   <property name="text">
    <string>English</string>
   </property>
   <property name="toolTip">
    <string>English Help</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionChinese">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Help.svg</normaloff>:/Menu Icon/Help.svg</iconset>
   </property>
   <property name="text">
    <string>Chinese</string>
   </property>
   <property name="toolTip">
    <string>Chinese Help</string>
   </property>
   <property name="shortcut">
    <string>Shift+H</string>
   </property>
  </action>
  <action name="actionGitHub">
   <property name="text">
    <string>GitHub</string>
   </property>
  </action>
  <action name="actionCSDN">
   <property name="text">
    <string>CSDN</string>
   </property>
  </action>
  <action name="actionStack_Overflow">
   <property name="text">
    <string>Stack Overflow</string>
   </property>
  </action>
  <action name="actionBilibili">
   <property name="text">
    <string>Bilibili</string>
   </property>
  </action>
  <action name="actionGitHub_2">
   <property name="text">
    <string>GitHub</string>
   </property>
  </action>
  <action name="actionCSDN_2">
   <property name="text">
    <string>CSDN</string>
   </property>
  </action>
  <action name="actionGitHub_Repository">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/GitHub.png</normaloff>:/Menu Icon/GitHub.png</iconset>
   </property>
   <property name="text">
    <string>GitHub Repository</string>
   </property>
  </action>
  <action name="actionCheck_Update">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Check Images.svg</normaloff>:/Menu Icon/Check Images.svg</iconset>
   </property>
   <property name="text">
    <string>Check Update</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Stop.svg</normaloff>:/Menu Icon/Stop.svg</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="actionVersion">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Version.png</normaloff>:/Menu Icon/Version.png</iconset>
   </property>
   <property name="text">
    <string>Version</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Import.svg</normaloff>:/Menu Icon/Import.svg</iconset>
   </property>
   <property name="text">
    <string>Import</string>
   </property>
  </action>
  <action name="actionConfiguration">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Route_Tool.svg</normaloff>:/Menu Icon/Route_Tool.svg</iconset>
   </property>
   <property name="text">
    <string>Configuration</string>
   </property>
  </action>
  <action name="actionCheck_Maze">
   <property name="icon">
    <iconset resource="Icons.qrc">
     <normaloff>:/Menu Icon/Preview.png</normaloff>:/Menu Icon/Preview.png</iconset>
   </property>
   <property name="text">
    <string>Check Maze</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="Icons.qrc"/>
 </resources>
 <connections/>
</ui>
